experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -482,18 +496,19 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 	private rebaseBranch(
 		branchHead: GraphCommit<TChange>,
 		onto: GraphCommit<TChange>,
-		repairData?: Map<RevisionTag, RepairDataStore>,
+		repairDataStoreProvider: IRepairDataStoreProvider,
```

Comment:
This argument must be the repair data store provider of the branch with head `branchHead`. If you collapse them into one argument (i.e. change `branchHead` to be `branch: SharedTreeBranch`) then you can avoid the possibility of ever passing the wrong one.